synthetics/test_conversions.go

```diff
@@ -214,3 +217,343 @@ func userInfoToMapSlice(obj *synthetics.V202101beta1UserInfo) []map[string]inter
 		"full_name": obj.FullName,
 	}}
 }
+
+// resourceDataToTest converts TF resource data to Test object. Only user-writable attributes are set.
+func resourceDataToTest(d *schema.ResourceData) (*synthetics.V202101beta1Test, error) {
+	test := synthetics.NewV202101beta1Test()
+
+	if id, ok := d.GetOk(idKey); ok {
```

Comment:
Is this `if <var> := <something>; <var> { ... }` a common Golang idiom? It looks unnecessarily obtuse to me (but I'd not call myself a Go programmer). Why not keep it simple?

```
id, ok := d.GetOk(idKey)
if ok {
    test.SetId(id.(string))
}
```

Reply:
Yes, it is common idiom, when function returns error or `ok` flag as second argument. I don't use it everywhere, but in such case IMO it still looks readable and makes the code more compact.